src/stashd/stash.rs

```diff
@@ -199,33 +199,58 @@ impl Stash for Runtime {
 
     fn merge(
         &mut self,
-        consignment: Consignment,
-    ) -> Result<Vec<Box<dyn Node>>, Error> {
-        let mut nodes: Vec<Box<dyn Node>> = vec![];
-        consignment.state_transitions.into_iter().try_for_each(
-            |(anchor, transition)| -> Result<(), Error> {
-                if self.storage.add_transition(&transition)? {
-                    nodes.push(Box::new(transition));
+        consignment: &Consignment,
+        known_seals: &Vec<OutpointReveal>,
+    ) -> Result<(), Error> {
+        // [PRIVACY]:
+        // Update transition data with the revealed state information that we
+        // kept since we did an invoice (and the sender did not know).
+        let reveal_known_seals =
+            |(_, assignments): (&usize, &mut Assignments)| match assignments {
+                Assignments::Declarative(_) => {}
+                Assignments::DiscreteFiniteField(set) => {
+                    *set = set
+                        .iter()
+                        .map(|a| {
+                            let mut a = a.clone();
+                            a.reveal_seals(known_seals.iter());
+                            a
+                        })
+                        .collect();
                 }
-                self.storage.add_anchor(&anchor)?;
-                self.indexer.index_anchor(&anchor)?;
```

Comment:
This is now missed. I know we were not adding int to the index before (since `merge` was never called) and the index is not actually used, but better to have this here for the times when we will use indexing.

Reply:
Added..